### ISSUE TITLE
[BUG] Filetype is not detected on setInputFiles

### ISSUE THREAD

**Issue by kobenguyent** [puppeteer/puppeteer#5543]
**Context:**

Puppeteer Version: 2.1.1
Operating System: MacOS

```
await el.setInputFiles('avatar.jpg')
await page.click('input[type=submit]');
```

Expected: 
`filetype to be image/jpeg`

Actual file:

```
{
  name: 'avatar.jpg',
  type: 'application/octet-stream',
  tmp_name: '/tmp/phpe1S4NZ',
  error: 0,
  size: 5621
}
```

The file is not empty but the file type is not detected. There is a second parameter of filetype in setInputFiles where I can specify a filetype. However, I assume this is a bug because the same test passes for older versions.

**Comment by uplg** [comment]
Related to : https://github.com/puppeteer/puppeteer/pull/5358 ?

Had same issues, enforced Pptr 2.0 and works ok, 2.1.0 introduce the mimeType bug

**Comment by kobenguyent** [comment]
> Related to : https://github.com/puppeteer/puppeteer/pull/5358 ?
> 
> 
> 
> Had same issues, enforced Pptr 2.0 and works ok, 2.1.0 introduce the mimeType bug

Yeah, it is. Thanks for pointing it out!

**Comment by DavertMik** [comment]
@mathiasbynens please could you take a look?
At CodeceptJS we run our tests across different engines and this is part of our test suite.
File detection works for WebDriver, Puppeteer 1.20 & Playwright but was broken since 2.0 and is still not fixed in 3.0.

**Comment by mathiasbynens** [comment]
@paullewis is gonna take a look at the remaining filepicker problems: #5654

**Comment by paullewis** [comment]
This should be fixed in 532ae573d29063ffd081ed9ee340d71b25320dec.

```javascript
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.goto('about:blank');
  await page.setContent('<input type="file">')

  const el = await page.evaluateHandle(() => document.querySelector('input'));
  await el.uploadFile('avatar.jpg')
  const type = await page.$eval('input', picker => picker.files[0].type);

  console.log(type);  // image/jpeg
  await browser.close();
})();
```
